pv_site_production/data/nwp_data_sources.py

```diff
@@ -0,0 +1,45 @@
+"""
+NWP Data sources
+"""
+import io
+from typing import Any
+
+import fsspec
+import xarray as xr
+from psp.data.data_sources.nwp import NwpDataSource as NwpDataSourceBase
+
+
+# Our original data source was designed with .zarr files in mind.
+# We adapt it to also support .netcdf files, in particular .netcdf files in the cloud.
+# TODO Implement this into the original NwpDataSource class instead.
+class NwpDataSource(NwpDataSourceBase):
+    """NWP Data source that supports both .zarr and .netcdf locally or in the cloud."""
+
+    def __init__(self, path: str, storage_kwargs: dict[str, Any] | None = None):
+        """Constructor"""
+        self._path = path
+
+        if storage_kwargs is None:
+            storage_kwargs = {}
+
+        self._storage_kwargs = storage_kwargs
+        self._open()
+
+    def _open(self):
+        path = self._path
+
+        if path.endswith(".zarr"):
+            return super()._open()
+        elif path.endswith(".netcdf"):
+
```

Comment:
worth adding some logging? Perhaps this could be in a different PR

Reply:
I'm not convinced we need logging in here, it's some pretty uneventful code!

Reply:
ok, no worries